internal/pkg/service/cli/cmd/cmd.go

```diff
@@ -302,36 +302,36 @@ func (root *RootCommand) printError(errRaw error) {
+				root.logger.InfofCtx(root.Context(), `Or use the "sync init" command in %s.`, dependencies.EmptyDir)
 			} else {
-				root.logger.Infof(`Please use %s.`, errDirNotFound.Expected())
+				root.logger.InfofCtx(root.Context(), `Please use %s.`, errDirNotFound.Expected())
 			}
 			if errDirNotFound.Expected() == dependencies.EmptyDir {
 				modifiedErrs.Append(errors.Wrapf(err, "directory is not empty"))
 			} else {
 				modifiedErrs.Append(errors.Wrapf(err, "neither this nor any parent directory is %s", errDirNotFound.Expected()))
 			}
 		case errors.Is(err, dependencies.ErrProjectManifestNotFound):
-			root.logger.Infof(`Project directory must contain the "%s" file.`, projectManifest.Path())
-			root.logger.Infof(`Please change working directory to a project directory.`)
-			root.logger.Infof(`Or use the "sync init" command in an empty directory.`)
+			root.logger.InfofCtx(root.Context(), `Project directory must contain the "%s" file.`, projectManifest.Path())
+			root.logger.InfofCtx(root.Context(), `Please change working directory to a project directory.`)
+			root.logger.InfofCtx(root.Context(), `Or use the "sync init" command in an empty directory.`)
 			modifiedErrs.Append(errors.Wrapf(err, `none of this and parent directories is project dir`))
 		case errors.Is(err, dependencies.ErrRepositoryManifestNotFound):
-			root.logger.Infof(`Repository directory must contain the "%s" file.`, repositoryManifest.Path())
-			root.logger.Infof(`Please change working directory to a repository directory.`)
-			root.logger.Infof(`Or use the "template repository init" command in an empty directory.`)
+			root.logger.InfofCtx(root.Context(), `Repository directory must contain the "%s" file.`, repositoryManifest.Path())
+			root.logger.InfofCtx(root.Context(), `Please change working directory to a repository directory.`)
+			root.logger.InfofCtx(root.Context(), `Or use the "template repository init" command in an empty directory.`)
 			modifiedErrs.Append(errors.Wrapf(err, `none of this and parent directories is repository dir`))
 		case errors.Is(err, dependencies.ErrTemplateManifestNotFound):
-			root.logger.Infof(`Template directory must contain the "%s" file.`, templateManifest.Path())
-			root.logger.Infof(`You are in the template repository, but not in the template directory.`)
-			root.logger.Infof(`Please change working directory to a template directory, for example "template/v1".`)
-			root.logger.Infof(`Or use the "template create" command.`)
+			root.logger.InfofCtx(root.Context(), `Template directory must contain the "%s" file.`, templateManifest.Path())
+			root.logger.InfofCtx(root.Context(), `You are in the template repository, but not in the template directory.`)
+			root.logger.InfofCtx(root.Context(), `Please change working directory to a template directory, for example "template/v1".`)
+			root.logger.InfofCtx(root.Context(), `Or use the "template create" command.`)
```

Comment:
Are you sure you want to always use the ctx methods? In cases like this logging the context values multiple times doesn't seem that great to me. But it's true that if we don't disallow the non-context methods completely it will be easy to forget to pass the context... I'm considering keeping the no context methods but renaming them:

```
logger.Infof(ctx, ...)
logger.InfofNoCtx(...)
```

Reply:
Will consider it later. :)